gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -403,6 +403,41 @@ message SetVariablesResponse {
   int64 key = 1;
 }
 
+message ModifyProcessInstanceRequest {
+  // the key of the process instance that should be modified
+  int64 processInstanceKey = 1;
+  // instructions describing which elements should be activated in which scopes,
+  // and which variables should be created
+  repeated ModificationActivateInstruction activateInstructions = 2;
+  // instructions describing which elements should be terminated
+  repeated ModificationTerminateInstruction terminateInstructions = 3;
+}
+
+message ModificationActivateInstruction {
+  // the id of the element that should be activated
+  string elementId = 1;
+  // the key of the ancestor scope the element instance should be created in
+  int64 ancestorElementInstanceKey = 2;
```

Comment:
🔧 Officially this field is optional, but we generally don't define that in the API. However, it might be useful to describe when it's needed or what is achieved by setting it to `-1`. For example, we did this with `version` of `CreateProcessInstanceRequest`.

Reply:
What's the reason we don't define that fields are optional in the API?

Reply:
Isn't setting something to -1 different than leaving the field out entirely?

Reply:
🤷 I have no clue why we don't actually, personally I don't mind it much